shadowsocks/client_test.go

```diff
@@ -244,3 +243,26 @@ func splitHostPortNumber(address string) (host string, port int, err error) {
 	}
 	return
 }
+
+func BenchmarkShadowsocksClient_UDPWrite(b *testing.B) {
+	proxyHost := "192.0.2.1"
+	proxyPort := 1
+	d, err := NewClient(proxyHost, proxyPort, testPassword, testCipher)
```

Comment:
Where do you run the server? What is it that you are testing?

Reply:
There is no server; this test is sending packets into a black hole.  It's benchmarking the overhead of constructing and sending a UDP packet, especially the memory allocations.

Reply:
I was also confused about this, could you please add a comment?

Reply:
Done